engine/src/flutter/impeller/renderer/backend/metal/context_mtl.mm

```diff
@@ -418,9 +418,20 @@ new ContextMTL(flags, device, command_queue,
     Lock lock(tasks_awaiting_gpu_mutex_);
     std::swap(tasks_awaiting_gpu, tasks_awaiting_gpu_);
   }
-  for (const auto& task : tasks_awaiting_gpu) {
-    task.task();
-  }
+  is_gpu_disabled_sync_switch_->Execute(
+      fml::SyncSwitch::Handlers()
+          .SetIfFalse([&] {
+            for (const auto& task : tasks_awaiting_gpu) {
+              task.task();
+            }
```

Comment:
Do you think we should acquire the lock inside the loop?  My concern is that if this takes too much time the OS might kill the process.

Reply:
hmm, yeah that is a good point.